content/getting-started/linux.md

```diff
@@ -0,0 +1,113 @@
+
+You can instead install the full source code to the TinyGo compiler itself, generally for people who wish to contribute to the project or want to build the compiler from sources.
+
+The third option is to use the Docker image. This has the benefit of making no changes to your system, however the image has a large download and installation size.
+
+## Quick Install
+
+If you are using Ubuntu or another Debian based Linux, download the DEB file from Github and install it using the following commands:
+
+```shell
+wget https://github.com/tinygo-org/tinygo/releases/download/v0.5.0/tinygo0.5.0.linux-amd64.deb
+sudo dpkg -i tinygo0.5.0.linux-amd64.deb
+```
+
+You will need to ensure that the path to the `tinygo` executable file is in your `PATH` variable.
+
+```shell
+export PATH=$PATH:/usr/local/tinygo/bin
+```
+
+You can test that the installation is working properly by running this code which should display the version number:
+
+```shell
+$ tinygo version
+tinygo version 0.5.0 linux/amd64
+```
+
+If you are only interested in compiling TinyGo code for WebAssembly then you are done with the installation.
+
+Otherwise, please continue with the installation of the additional requirements for your desired microcontroller.
+
+### Additional Requirements for Microcontrollers
+
+There are some additional requirements to compile TinyGo programs that can run on microcontrollers.
+
+#### ARM Cortex-M
+
+To compile TinyGo programs for ARM based processors you must install CLang 8 (`clang-8`) for building assembly files and the compiler runtime library [https://compiler-rt.llvm.org/](https://compiler-rt.llvm.org/).
+
+Some boards also require a special flashing tool for that particular chip, like `openocd` or `nrfjprog`. See the documentation page for your board as listed [here](../../microcontrollers/) to see which flashing tool is required for your target board.
+
+#### AVR (Arduino)
+
+To compile and flash TinyGo programs for AVR based processors you must install some extra tools:
+
+```shell
+sudo apt-get install avr-gcc
+sudo apt-get install avr-libc
+sudo apt-get install avrdude
+```
+
+This should allow you to compile and flash TinyGo programs on an Arduino or other supported AVR-based board.
+
+## Source Install
+
+You can install TinyGo from source, which includes installing the LLVM compiler toolchain.
+
+First, obtain the TinyGo source code:
+
+```shell
+go get -d github.com/tinygo-org/tinygo
+cd $GOPATH/src/github.com/tinygo-org/tinygo
+```
+
+Once you have the code, you can install the various prerequisites using [`dep`](https://golang.github.io/dep/):
+
+```shell
+dep ensure
+```
+
+Now you can run the following make tasks to download and build the LLVM toolchain. Please note that this is likely to take at least 1 hour even on a fast machine.
+
+```shell
+make llvm-source
+make llvm-build
+```
+
+Once the LLVM toolchain is installed, you can build the TinyGo binary that is linked to local system libraries like this:
+
+```shell
+make tinygo
```

Comment:
This performs an install by dynamically linking (`go install`), use `make static` to build a static binary.
Perhaps the Makefile should be changed here. I don't really use `make tinygo` myself, for testing I almost always run `go install` except when I want to statically, in which case I run `make static`. (see also https://github.com/tinygo-org/tinygo/pull/281).

Reply:
Updated based on latest build info.